### PR TITLE
feat: add @sapphire/embed-jsx

### PR DESCRIPTION
**Pull request by PyroTechniac**

### CODE REVIEW

File: packages/embed-jsx/package.json
Comment (Favna) [inline packages/embed-jsx/package.json:3]:
```suggestion
	"version": "0.0.1",
```

Lerna versioning stupidity but this is needed

File: packages/embed-jsx/package.json
Comment (Favna) [inline packages/embed-jsx/package.json:14]: Please change to match other packages. We give credit through all contributors and if you so choose we can also attribute this part of the code to you as code owners.

File: packages/embed-jsx/package.json
Comment (Favna) [inline packages/embed-jsx/package.json:26]: Remove this

File: packages/embed-jsx/package.json
Comment (Favna) [inline packages/embed-jsx/package.json:30]: This is not the right repo

File: packages/embed-jsx/package.json
Comment (Favna) [inline packages/embed-jsx/package.json:33]: Other packages without tests do not edit. See those.

File: packages/embed-jsx/package.json
Comment (Favna) [inline packages/embed-jsx/package.json:1]: Overall equalise to other packages package.json files

File: packages/embed-jsx/tsconfig.json
Comment (Favna) [inline packages/embed-jsx/tsconfig.json:4]: Remove this line

File: packages/embed-jsx/README.md
Comment (Favna) [inline packages/embed-jsx/README.md:1]: You can look at other packages to at least apply the structure

**Review by Favna — DISMISSED** [review]
Also needs updates:

README in root to mention the package in badges

- typedoc.json to filter index files
- .all-contributorsrc to update the new README

**Comment by Nytelife26** [comment]
Fix lint errors and apply the general package structure like Favna suggested, and I'll review :)

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (Nytelife26) [inline packages/embed-jsx/src/lib/EmbedJsx.ts:46]: Can you add typedoc annotations please?

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (Nytelife26) [inline packages/embed-jsx/src/lib/EmbedJsx.ts:3]:
```suggestion
const EMBED_TYPES = ['embed', 'title', 'field', 'timestamp', 'footer', 'description', 'url', 'image'];
```
TypeScript can infer that it is a constant, no need for the casting

**Review by Nytelife26 — APPROVED** [review]
LGTM aside from this. Note that this is a code review only, so this is also subject to Favna's qualms.

**Review by Nytelife26 — CHANGES_REQUESTED** [review]
Meant to request changes, not approve. My bad.

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (PyroTechniac) [reply packages/embed-jsx/src/lib/EmbedJsx.ts:3]: removing this line prevents ts from inferring all the switch cases, which makes it think that `make` isn't exhaustive

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (PyroTechniac) [reply packages/embed-jsx/src/lib/EmbedJsx.ts:46]: Docs will be done once i can finish up the readme

**Comment by PyroTechniac** [comment]
Should be ready for review, unless people want more features :eyes:

File: packages/embed-jsx/package.json
Comment (Nytelife26) [inline packages/embed-jsx/package.json:42]:
```suggestion
		"discord.js": "^12.5.3"
```
Unless this particular version adds something required in the util, I'd suggest making it `12.x`.

**Review by Nytelife26 — CHANGES_REQUESTED** [review]
LGTM aside from this last nitpick now

File: packages/embed-jsx/package.json
Comment (kyranet) [reply packages/embed-jsx/package.json:42]: Older versions are discouraged as they not longer work.

Comment (PyroTechniac) [reply packages/embed-jsx/package.json:42]: any discord.js version that has a MessageEmbed structure that matches the current one should work fine, as that's the only import

**Review by Favna — APPROVED** [review]

**Review by Favna — DISMISSED** [review]

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (vladfrangu) [inline packages/embed-jsx/src/lib/EmbedJsx.ts:3]: author and thumbnail are missing

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (vladfrangu) [inline packages/embed-jsx/src/lib/EmbedJsx.ts:77]: Can't you unite these? `info.shift() ?? {}`?

File: tsconfig.dev.json
Comment (vladfrangu) [inline tsconfig.dev.json:7]: I don't think these should be here?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (PyroTechniac) [reply packages/embed-jsx/src/lib/EmbedJsx.ts:3]: I have no idea how to do author cause of all the options, suggestions welcome

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (PyroTechniac) [reply packages/embed-jsx/src/lib/EmbedJsx.ts:77]: will the `shift` still occur if the value is null?

File: tsconfig.dev.json
Comment (PyroTechniac) [reply tsconfig.dev.json:7]: cc: @Favna

**Comment by PyroTechniac** [comment]
before this is merged I need to combine title and url into one as a url isn't a valid option without a title

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (vladfrangu) [reply packages/embed-jsx/src/lib/EmbedJsx.ts:77]: image

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (vladfrangu) [reply packages/embed-jsx/src/lib/EmbedJsx.ts:3]:
```jsx
<author iconURL="https://owo.cat">
Hi MOM!
</author>
```

```jsx
<Thumbnail url="oof" />
```

File: tsconfig.dev.json
Comment (Favna) [reply tsconfig.dev.json:7]: Without these the tests in CI fail because they use this config file @vladfrangu @PyroTechniac

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (PyroTechniac) [reply packages/embed-jsx/src/lib/EmbedJsx.ts]: Fixed in e893298837191db906af815efd1562aa8a5e6ae6

File: packages/embed-jsx/src/lib/EmbedJsx.ts
Comment (PyroTechniac) [reply packages/embed-jsx/src/lib/EmbedJsx.ts]: Fixed in 4edaf9bc198e25e4d4222c33a1d7e5b891cb338a

File: packages/embed-jsx/README.md
Comment (Favna) [inline packages/embed-jsx/README.md:3]:
```suggestion
![Sapphire Logo](https://cdn.skyra.pw/gh-assets/sapphire-banner.png)
```

We changed the banner

**Review by Favna — CHANGES_REQUESTED** [review]

**Review by Favna — DISMISSED** [review]

**Review by Favna — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]
Looks fine

**Pull request merged**
